src/dep/phonetics/double_metaphone.c

```diff
@@ -961,6 +961,7 @@ void DoubleMetaphone(const char *str, char **primary_pp, char **secondary_pp) {
   if (primary_pp) {
     if (*primary_pp) {
       free(*primary_pp);
+      *primary_pp = NULL;
```

Comment:
I probably missed it in the big leak fixes PR but I do not think the Phonetic algorithm should be responsible to free the primary_pp as it is an out variable. Also how come its not NULL from the first place?

Reply:
I'll need to see why we don't clear this externally. But let's handle that later.